eth/utils/state.py

```diff
@@ -1,14 +1,50 @@
+from typing import (
+    Dict,
+    Iterable,
+    Tuple,
+    Union,
+)
+
+from mypy_extensions import (
```

Comment:
Based on what you choose from the [stated options](https://github.com/ethereum/py-evm/pull/1404#issuecomment-431346890) you may need to add an `if TYPECHECKING` here to only do that import under `mypy` runs.